mypy/checkmember.py

```diff
@@ -555,13 +556,15 @@ def analyze_var(name: str,
                 dispatched_type = meet.meet_types(mx.original_type, itype)
                 functype = check_self_arg(functype, dispatched_type, var.is_classmethod,
                                           mx.context, name, mx.msg)
-                signature = bind_self(functype, mx.self_type, var.is_classmethod)
+                signature = freshen_function_type_vars(functype)
```

Comment:
I think we should freshen the function type vars before the check_self_arg, to match how it is done elsewhere

Reply:
Done, thanks for the pointer.